.golangci.yml

```diff
@@ -9,95 +9,54 @@ run:
   modules-download-mode: readonly
 
 linters:
-  enable:
-    - bodyclose
-    - deadcode
-    - depguard
-    - dogsled
-    - dupl
-    - errcheck
-    - exportloopref
-    - funlen
-    # We use init() functions.
-    #- gochecknoinits
-    - goconst
-    - gocritic
-    - gocyclo
-    - gofmt
-    - goimports
-    # This is stupid since it reports every constant as magic-number.
-    #- gomnd
-    - goprintffuncname
-    - gosec
-    - gosimple
-    - govet
-    - ineffassign
-    # This thing reports long lines, but we have a lot of them…
-    #- lll
-    - misspell
-    # We often just have a 'return' when using named returns.
-    #- nakedret
-    - nolintlint
-    - rowserrcheck
-    - staticcheck
-    - structcheck
-    - stylecheck
-    - typecheck
-    - unconvert
-    - unparam
-    - unused
-    - varcheck
-    - whitespace
-    - asciicheck
-    # We often use globals for backend stuff
-    #- gochecknoglobals
-    - gocognit
-    - godot
-    # Reports TODO/BUG/FIXME
-    #- godox
-    # Keeps reporting 'do not define dynamic errors, use wrapped static errors instead'.
-    #- goerr113
-    # We don't care about alignment right now.
-    #- maligned
-    - nestif
-    - prealloc
-    # This always reports that tests should be Black Box.
-    #- testpackage
-    #- wsl
+  enable-all: true
+  disable:
+    # Up for discussion:
+    - gofumpt # Pedantic formatting
+    - revive # go lint drop in
+    - ifshort # we often dont use `if err := …` for readability.
+    - tparallel # We dont always use parallel tests.
+
+    - wrapcheck # We do not rewrap all errors from other packages.
+    - varnamelen # We often use single letter arguments.
+    - exhaustivestruct # We often have uninitialized fields of structs.
+    - errorlint # We have interface errors which an impl needs to return.
+    - nakedret # We use naked returns in long functions.
+    - lll # We use long lines.
+    - gochecknoglobals # We have a lot of globals.
+    - nlreturn # "new line before return"; not useful.
+    - paralleltest # We like parallel tests.
+    - ireturn # We often return interfaces.
+    - wsl # Formatting disabled for now.
+    - gochecknoinits # We use init functions.
+    - gci # We have our own import order
+    # Unimportant
+    # Deprecated
+    - maligned
+    - scopelint
+    - golint
+    - interfacer
+
+linters-settings:
+  gocritic:
+    disabled-checks:
+      # Accept code in the form of object.Call(object), eg:
+      # assert.True(t, null.Equals(null), "Expected equality of zero address to itself")
```

Comment:
Why would we want to allow that in general? This is an instance where I think it makes sense to disable the linter for a specific line.

Reply:
Ok i updated it.